### PR TITLE
feat(addon-table): add directive to sort by order

### PR DESCRIPTION
**Pull request by zarghamkhandev**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?
To specify sort direction one needs to pass direction as 1 or -1 which is a bit confusing.

Closes # https://github.com/Tinkoff/taiga-ui/issues/2266

## What is the new behavior?
Allows user to specify direction in terms of order e.g "asc" or "desc" instead of 1 or -1.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by zarghamkhandev** [comment]
regarding docs, I am not sure if should i add this to existing examples or create a new example to demonstrate the directive.

File: projects/addon-table/components/table/directives/direction-by-order.directive.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import {Directive, Inject, Input, Output} from '@angular/core';
+import {map} from 'rxjs/operators';
+
+import {TuiTableDirective} from './table.directive';
+
+@Directive({
+    selector: `table[tuiTable][tuiDirectionByOrder]`,
+})
+export class TuiDirectionByOrderDirective<T> {
```
Comment (splincode) [inline projects/addon-table/components/table/directives/direction-by-order.directive.ts:9]: What do you think, will you be able to add a unit test or a demo example?

**Review by splincode — APPROVED** [review]

File: projects/addon-table/components/table/directives/direction-by-order.directive.ts (hunk 1 above)
Comment (zarghamkhandev) [reply projects/addon-table/components/table/directives/direction-by-order.directive.ts:9]: yes, will do both :) will update soon

File: projects/addon-table/components/table/directives/direction-by-order.directive.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import {Directive, Inject, Input, Output} from '@angular/core';
+import {map} from 'rxjs/operators';
+
+import {TuiTableDirective} from './table.directive';
+
+@Directive({
+    selector: `table[tuiTable][tuiDirectionByOrder]`,
```
Comment (waterplea) [inline projects/addon-table/components/table/directives/direction-by-order.directive.ts:7]: Maybe just `tuiDirectionOrder`? And the directive name/file name as well.

**Review by waterplea — APPROVED** [review]

**Pull request closed without merging**